packages/jsii-java-runtime/project/src/main/java/software/amazon/jsii/JsiiEngine.java

```diff
@@ -578,6 +576,35 @@ public JsiiObjectRef createNewObject(final Object uninitializedNativeObject, fin
         return overrides.values();
     }
 
+    /**
+     * Crawls up the inheritance tree of {@code classToInspect} in order to determine the jsii-visible
+     * interfaces that are implemented, using a procedure akin to:
+     *
+     * <ol>
+     *  <li>If {@code classToInspect} is annotated with @Jsii and is an interface, return a singleton set with the FQN</li>
+     *  <li>If {@code classToInspect} is annotated with @Jsii and is NOT an interface, return an empty set</li>
```

Comment:
If the class annotated with JSII and is not an interface, but it implements interfaces, don't we want to return the FQNs of the interfaces it implements? What am I missing?

Reply:
Those interfaces are implemented by the JSII-known type, so there is no point in re-declaring them (base type implies interfaces).